infrastructure/api/api.go

```
@@ -0,0 +1,180 @@
+func (u *UrbanDictionary) GetDefinitions(term string) (*model.List, error) {
+	var list *model.List
+	url := fmt.Sprintf("%s?term=%s", u.ApiURL, term)
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	request.Header.Add(rapidapiKeyName, u.Headers[rapidapiKeyName])
+
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	err = errorStatus(response.StatusCode)
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(body, &list)
+	if err != nil {
+		return nil, err
+	}
+	if len(list.Definitions) == 0 {
+		return nil, model.ErrNotFound{Term: term}
+	}
+	return list, nil
+}
+
+// GetDefinitionById reads a local csv to find the definition by id paramater
+func (u *UrbanDictionary) GetDefinitionById(id string) (*model.List, error) {
+	list := new(model.List)
+
+	definitions, err := u.Read(id)
+	if err != nil {
+		return nil, err
+	}
+	if len(definitions) == 0 {
+		return nil, model.ErrNotFoundInCSV{Id: id}
+	}
+	list.Definitions = definitions
+
+	return list, nil
+}
+
+// Open returns a pointer of the local csv file
+func (u *UrbanDictionary) Open() (*os.File, error) {
+	file, err := os.OpenFile(u.CSVPath, os.O_CREATE|os.O_APPEND|os.O_RDWR, os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+	return file, nil
+}
+
+// Read takes every definition record from the csv file into a []Definition
+func (u *UrbanDictionary) Read(id string) ([]*model.Definition, error) {
+	definitions := make([]*model.Definition, 0)
+	file, err := u.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	csvReader := csv.NewReader(file)
+	for {
+		line, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			fmt.Println(err.Error())
+			continue
+		}
+		if id == line[0] {
+			idCsv, _ := strconv.Atoi(line[0])
```

Don't ignore this error, handle it